cmd/query/app/apiv3/http_gateway.go

```diff
@@ -179,21 +186,26 @@ func (h *HTTPGateway) findTraces(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	traces, err := h.QueryService.FindTraces(r.Context(), queryParams)
+	findTracesIter := h.QueryService.FindTraces(r.Context(), *queryParams)
+	traces, err := iter.FlattenWithErrors(findTracesIter)
```

Comment:
from this point it's the same as in FindTrace, I would move into `returnTraces` helper